lib/auth/default_current_user_provider.rb

```diff
@@ -412,6 +412,14 @@ def parameter_api_patterns
   # However, in some scenarios it is essential to send them via url parameters
   # so we need to add some exceptions
   def api_parameter_allowed?
+    if @env[ActionDispatch::Http::Parameters::PARAMETERS_KEY].nil? && @request.path_info.present?
+      @env[ActionDispatch::Http::Parameters::PARAMETERS_KEY] = begin
```

Comment:
Discussed this internally and we want to avoid setting a header that Rails relies on internally. Instead, we will store it in a custom Discourse header instead and pass the params through the the route matcher.